app/js/views/themes-new.jsx

```diff
@@ -146,104 +144,125 @@ export default class ThemesNew extends BaseView {
   }
 
   renderBody() {
-    let optionNodes;
+    let headerClassName = 'new-theme__header';
+    if (this.state.selectedGetterValueIndex < 0) {
+      headerClassName += ' new-theme__header--hidden';
+    }
+
+    return (
+      <div className="app-view__fill-body new-theme">
+        <h2 className="new-theme__header">If</h2>
+        {this.renderGetterSelector()}
+        {this.renderGetterValueSelector()}
 
+        <h2 className={headerClassName}>Do</h2>
+        {this.renderSetterSelector()}
+        {this.renderSetterValueSelector()}
+      </div>
+    );
+  }
+
+  renderGetterSelector() {
     let className = 'new-theme__select';
-    if (this.state.serviceWithGet !== null) {
+    if (this.state.selectedGetterIndex >= 0) {
       className += ' new-theme__select--selected';
     }
-    optionNodes = this.state.servicesWithGet.map((service, id) => (
-      <option key={id} value={id}>{service.label}</option>
+
+    const optionNodes = this.state.getters.map((getter, index) => (
+      <option key={index} value={index}>{getter.name}</option>
     ));
-    let serviceSelector = (
-      <select value={this.state.serviceWithGet}
-              onChange={this.handleServiceWithGetChange}
+
+    return (
+      <select value={this.state.selectedGetterIndex}
+              onChange={this.onGetterSelected}
```

Comment:
nit: In other JSX files, event listeners are called on the model of `handle...()`. Do you prefer the `on...()` model?